darwin/future/core/types/query.py

```diff
@@ -106,47 +110,38 @@ def __len__(self) -> int:
             return 0
         return len(self.filters)
 
-    def __iter__(self) -> Query[T]:
+    def __iter__(self) -> Query[T, R]:
         self.n = 0
         return self
 
-    def __next__(self) -> QueryFilter:
-        if self.filters is None:
-            self.filters = []
-        if self.n < len(self.filters):
-            result = self.filters[self.n]
+    def __next__(self) -> R:
+        if self.results is None:
+            self.results = list(self.collect())
```

Comment:
Interaction here becomes
self.collect() -> MetaObject
Then MetaObject stores a list of R and implement iteration dunders